src/main/java/com/google/sps/servlets/NaturalLanguageServlet.java

```diff
@@ -63,9 +95,29 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         if (this.nlp == null) {
             this.nlp = new NaturalLanguageProcessor();
         }
+        if (this.db == null) {
+            this.db = new DatabaseImpl();
+        }
 
         // Builds the Java object from JSON and preprocesses the captions by redefining time ranges
         YoutubeCaptions youtubeCaptions = gson.fromJson(json, YoutubeCaptions.class);
+
```

Comment:
can we initialize metadataList (line 141) here (or at least at the beginning of the doPost function)? might need to add things like error messages beforehand.

Reply:
yea, that should be fine